### ISSUE TITLE
[HOLD for payment 2024-07-22] [HOLD for payment 2024-07-17] [Travel] [Refactor] Create a new shared component for AddressPage

### ISSUE THREAD

**Issue by mountiny** [Expensify/App#41965]
### Problem

Currently `WorkspaceProfileAddressPage` and `AddressPage` have some repeated code (ex. handleAddressChange callback, useEffect on countryFromUrl or useEffect on address)

Moreover, `AddressPage` has code which has no effect - ex. useEffect on stateFromUrl (because stateFromUrl is always undefined).

Those code readability issues were detected while creating and reviewing `WorkspaceProfilePage` PR ex. here or here.

### Solution
We should create a new component as a middle layer between AddressPage/WorkspaceProfileAddressPage and AddressForm. Current AddressPage should be renamed to PersonalAddressPage and the new component should be named AddressPage.

New `AddressPage` should contain all the common logic for `PersonalAddressPage` and `WorkspaceAddressPage`.

`WorkspaceAddressPage` and `PersonalAddressPage` should only contain logic behind callbacks to API.
Those are rather simple changes which should improve code readability and make it easier to maintain.

<details><summary>Issue Owner</summary>Current Issue Owner: @slafortune / @joekaufmanexpensify</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @slafortune</details>

**Comment by smelaa** [comment]
Hi, I am Aleksandra Smela from SWM and I would like to work on this issue!

**Comment by dubielzyk-expensify** [comment]
I've been assigned as a design reviewer. Would you like me to have a look at [this PR](https://github.com/Expensify/App/pull/41970) now or is it still too early?

**Comment by mountiny** [comment]
I dont think this should have any change in design so maybe its not required and you can unassign. Its automated as its a new feature.

The code changes should be unnoticed in product, purely code readability

**Comment by slafortune** [comment]
Looks like changes in the PR have been approved and it's moving forward!  I'll be on vacation until June 4th, so adding another BZ in the meantime.

**Comment by mountiny** [comment]
Asked to get this PR merged

**Comment by joekaufmanexpensify** [comment]
>@mountiny could you assign me here? thanks

Done!

**Comment by mountiny** [comment]
PR was reverted so we are going to give this a second try

**Comment by mountiny** [comment]
Going to have to handle this regression in the next round https://github.com/Expensify/App/issues/42638

**Comment by mateuuszzzzz** [comment]
Hi, I'm Mateusz Rajski from SWM and I would like to work on this issue

**Comment by mateuuszzzzz** [comment]
Hi! I'm Mateusz Rajski from SWM and I would like to work on this issue.

**Comment by joekaufmanexpensify** [comment]
@mountiny should we have @mateuuszzzzz handle this?

**Comment by mountiny** [comment]
Yes, @mateuuszzzzz what is your ETA for the new PR?

**Comment by mateuuszzzzz** [comment]
> Yes, @mateuuszzzzz what is your ETA for the new PR?

Doesn't seem like something complicated, the code is practically ready, we just need to fix source of regression. Nevertheless, I will investigate it today/tomorrow and give better estimation.

**Comment by mateuuszzzzz** [comment]
@mountiny  I should be able to prepare PR on friday

**Comment by slafortune** [comment]
Updating the title since we'll need to wait on a fix/pr on this - payment isn't due at this time.

**Comment by mateuuszzzzz** [comment]
@mountiny I got a fix for country input, but I noticed we might have more bugs there.

The [Issue](https://github.com/Expensify/App/issues/42638) describes a problem with the country input, but I am also having difficulty setting other inputs to an empty string (on the main branch).

[This if](https://github.com/Expensify/App/blob/9c5bf210080722a0f7a929a0e0d3ab2e49922f70/src/pages/settings/Profile/PersonalDetails/AddressPage.tsx#L79) should clear State, City and Zip / Postcode fields if country is updated.

Could you confirm whether this is an additional bug? I can address it within the same PR, although I will need to do a more thorough investigation of this new issue.

https://github.com/Expensify/App/assets/60450261/46749bf3-6f01-4bc6-83b7-1717b782a18e

**Comment by mountiny** [comment]
@mateuuszzzzz Yeah, I imagine, though, should also be cleared.

But if this is happening in `main` we do not have necessarily address it here

**Comment by nkdengineer** [comment]
@mateuuszzzzz There was [this proposal](https://github.com/Expensify/App/issues/42638#issuecomment-2133822796) that fixes [the regression](https://github.com/Expensify/App/issues/42638), in case it's helpful to you.

**Comment by zfurtak** [comment]
Hello there :) If it's okay I will be working on this issue instead of @mateuuszzzzz as recently he's had a lot stuff to do with Hybrid app.

**Comment by slafortune** [comment]
@zfurtak was there a conversation about this somewhere you can link here?

**Comment by mountiny** [comment]
@slafortune I think this is fine for SWM to distribute internally. I will assign @zfurtak to this issue now

**Comment by mountiny** [comment]
There is one sneaky regression but I think its really hard to catch in this flow so I suggest a normal $250 to @getusha and we can close this issue

**Comment by slafortune** [comment]
Based on this [comment](https://github.com/Expensify/App/issues/41965#issuecomment-2220755499)
@getusha is due for the C+ role $250 via newdot 👍 
@zfurtak and @mateuuszzzzz are expert contributors and not due payment.
